python/dgl/distributed/sparse_emb.py

```diff
@@ -136,6 +139,15 @@ def step(self):
                     # We'll need to scale them with the state sum on the kvstore server
                     # after we push them.
                     grads = F.cat(grads, 0)
-                    kvstore.push(name, idxs, grads)
+
+                uniq_idxs, inverse_idxs = F.unique(idxs, return_inverse=True)
```

Comment:
If it is a distributed system, shall we gather grad from all machines?

Reply:
this requires significant modification. let's have this for now. we need to test if it's necessary to merge all gradients first.